arduino/cores/packagemanager/package_manager.go

```diff
@@ -93,7 +93,12 @@ func (pmb *Builder) BuildIntoExistingPackageManager(target *PackageManager) {
 	target.tempDir = pmb.tempDir
 	target.packagesCustomGlobalProperties = pmb.packagesCustomGlobalProperties
 	target.profile = pmb.profile
-	target.discoveryManager = pmb.discoveryManager
+	if target.discoveryManager != nil {
+		target.discoveryManager.Clear()
+	} else {
+		target.discoveryManager = pmb.discoveryManager
+	}
+	target.discoveryManager.AddAllDiscoveriesFrom(pmb.discoveryManager)
```

Comment:
`target.discoveryManager` can't be `nil` by construction

```suggestion
	target.discoveryManager.Clear()
	target.discoveryManager.AddAllDiscoveriesFrom(pmb.discoveryManager)
```

Reply:
I tried to remove it but got a `runtime error: invalid memory address or nil pointer dereference` on the first initialization

Reply:
The empty target is coming from https://github.com/arduino/arduino-cli/blob/master/arduino/cores/packagemanager/package_manager.go#L107